lib/Backend/arm64/LowerMD.cpp

```diff
@@ -1392,47 +1416,75 @@ LowererMD::LowerEntryInstr(IR::EntryInstr * entryInstr)
             {
-                regPushedSize += MachRegInt;
-                IR::Instr * instrStore = IR::Instr::New(Js::OpCode::STR, this->m_func);
-                instrStore->SetDst(IR::IndirOpnd::New(IR::RegOpnd::New(nullptr, RegSP, TyMachReg, this->m_func), -regPushedSize, TyMachReg, this->m_func));
-                instrStore->SetSrc1(IR::RegOpnd::New(reg, TyMachReg, this->m_func));
-                insertInstr->InsertBefore(instrStore);
+                if (src1 == nullptr)
+                {
+                    src1 = IR::RegOpnd::New(reg, TyMachReg, this->m_func);
+                }
+                else
+                {
+                    Assert(src2 == nullptr);
+                    src2 = IR::RegOpnd::New(reg, TyMachReg, this->m_func);
+                }
+            }
+
+            if (src1 != nullptr && src2 != nullptr)
+            {
+                IR::Instr * instrStp = IR::Instr::New(Js::OpCode::STP_PRE,
+                    IR::IndirOpnd::New(IR::RegOpnd::New(nullptr, RegSP, TyMachReg, this->m_func), -(2 * MachRegInt), TyMachReg, this->m_func),
+                    src2, src1, this->m_func);
+                insertInstr->InsertBefore(instrStp);
+                src1 = src2 = nullptr;
             }
         }
+
+        // NOTE - We made sure that usedRegs is even
+        Assert(src1 == nullptr);
+        Assert(src2 == nullptr);
     }
 
     if (!usedDoubleRegs.IsEmpty())
     {
+        IR::RegOpnd* src1 = nullptr;
+        IR::RegOpnd* src2 = nullptr;
         for (RegNum reg = LAST_CALLEE_SAVED_DBL_REG; reg >= FIRST_CALLEE_SAVED_DBL_REG; reg = (RegNum)(reg - 1))
         {
             Assert(LinearScan::IsCalleeSaved(reg));
             if (usedDoubleRegs.Test(RegEncode[reg]))
             {
-                regPushedSize += MachRegDouble;
-                IR::Instr * instrStore = IR::Instr::New(Js::OpCode::FSTR, this->m_func);
-                instrStore->SetDst(IR::IndirOpnd::New(IR::RegOpnd::New(nullptr, RegSP, TyMachReg, this->m_func), -regPushedSize, TyMachReg, this->m_func));
-                instrStore->SetSrc1(IR::RegOpnd::New(reg, TyMachReg, this->m_func));
-                insertInstr->InsertBefore(instrStore);
+                if (src1 == nullptr)
+                {
+                    src1 = IR::RegOpnd::New(reg, TyMachReg, this->m_func);
+                }
+                else
+                {
+                    Assert(src2 == nullptr);
+                    src2 = IR::RegOpnd::New(reg, TyMachReg, this->m_func);
+                }
+            }
+
+            if (src1 != nullptr && src2 != nullptr)
+            {
+                IR::Instr * instrStp = IR::Instr::New(Js::OpCode::STP_PRE,
+                    IR::IndirOpnd::New(IR::RegOpnd::New(nullptr, RegSP, TyMachReg, this->m_func), -(2 * MachRegDouble), TyMachReg, this->m_func),
+                    src2, src1, this->m_func);
+                insertInstr->InsertBefore(instrStp);
+                src1 = src2 = nullptr;
```

Comment:
Ugh, this isn't going to work with the way the unwind codes operate. You have to pre-allocate the space then use FSTP (no predecrement) with a positive offset to store them.